liquigraph-core/src/main/java/org/liquigraph/core/io/GraphJdbcConnector.java

```diff
@@ -10,12 +10,27 @@
 
 public class GraphJdbcConnector implements LiquigraphJdbcConnector {
 
+    private final LockGuardian lockGuardian;
+
+    public GraphJdbcConnector(LockGuardian lockGuardian) {
+        this.lockGuardian = lockGuardian;
+    }
+
+    /**
+     * Acquires a new connection to the configured instance
+     * and tries to lock it (fail-fast).
+     *
+     * @see {@link LockGuardian}
+     * @param configuration Liquigraph settings
+     * @return JDBC connection
+     */
     @Override
     public final Connection connect(Configuration configuration) {
         try {
             Class.forName("org.neo4j.jdbc.Driver");
             Connection connection = DriverManager.getConnection(makeUri(configuration));
             connection.setAutoCommit(false);
+            lockGuardian.lock(connection);
```

Comment:
Yep, and the shutdown hook is only a good fit for Maven/Shell users, not core API ones.
I'll have to review this.